Source/JavaScriptCore/runtime/JSGlobalObject.h

```diff
@@ -587,8 +587,10 @@ class JSGlobalObject : public JSSegmentedVariableObject {
 #endif
 
     bool m_evalEnabled { true };
+    bool m_trustedEvalEnabled { true };
     bool m_webAssemblyEnabled { true };
     bool m_requiresTrustedTypes { true };
+    bool m_requiresTrustedTypesEnforced { true };
```

Comment:
This seems like peculiar wording: "requires trusted types enforced" doesn’t seem like idiomatic language. Maybe we could find a way to word this more the way a person would say it in conversation, or in a way that matches what the specification calls it.

Reply:
Yeah, these bools should be an `enum` that conveys the TrustedTypes enforcement status. It seems like it could also contain the m_trustedEvalEnabled flag too.

Reply:
Okay yeah a three value enum for None, ReportOnly and Enforced makes sense. I'll update accordingly.

Not sure if including trusted types eval will work but I'll take a look when doing the change.

Reply:
It seems like `trustedEvalEnabled == true` only matters when `requiresTrustedTypesEnforced == true` so it seems like you could just make a fourth enum value `EnforcedWithEvalEnabled`.

Reply:
You were correct I forgot they were dependent on each other in all usages. So I've swapped this for a 4 value enum.

Reply:
So I've used the enum at the getter places too because I found it to be clearer to read at the call site. Happy to change to boolean getters if that's prefered.

Reply:
If you think it’s clearer at the call sites, we should keep it that way. That’s what matters.

I am surprised to hear that you think the enumeration is clearer than a boolean getter, though; did you actually consider both?

Reply:
Yeah I had it as the booleans originally and changed it to just evaluate at the callsite with the enum values. Naming them is difficult given the different states, with the enum it just makes it clear which state I'm checking.

I'm happy to defer if others disagree though. I can be persuaded either way.